src/App.tsx

```diff
@@ -1,16 +1,24 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setNewMovies] = useState(moviesFromServer);
+
+  const handleAdd = (newMovie: Movie) => {
+    setNewMovies([...movies, newMovie]);
```

Comment:
use functional setState here
```suggestion
    setNewMovies((currentMovies) => [...currentMovies, newMovie]);
```